finality-provider/service/chain_poller.go

```diff
@@ -168,41 +202,65 @@ func (cp *ChainPoller) pollChain() {
 
 	cp.waitForActivation()
 
+	ticker := time.NewTicker(cp.cfg.PollInterval)
+	defer ticker.Stop()
+
 	var failedCycles uint32
 
 	for {
-		// start polling in the first iteration
-		blockToRetrieve := cp.nextHeight
-		block, err := cp.blockWithRetry(blockToRetrieve)
+		latestBlock, err := cp.getLatestBlockWithRetry()
 		if err != nil {
 			failedCycles++
 			cp.logger.Debug(
-				"failed to query the consumer chain for the block",
+				"failed to query the consumer chain for the latest block",
 				zap.Uint32("current_failures", failedCycles),
-				zap.Uint64("block_to_retrieve", blockToRetrieve),
 				zap.Error(err),
 			)
 		} else {
-			// no error and we got the header we wanted to get, bump the state and push
-			// notification about data
-			cp.nextHeight = blockToRetrieve + 1
-			failedCycles = 0
-			cp.metrics.RecordLastPolledHeight(block.Height)
-
-			cp.logger.Info("the poller retrieved the block from the consumer chain",
-				zap.Uint64("height", block.Height))
-
-			// push the data to the channel
-			// Note: if the consumer is too slow -- the buffer is full
-			// the channel will block, and we will stop retrieving data from the node
-			cp.blockInfoChan <- block
+			// start polling in the first iteration
+			blockToRetrieve := cp.NextHeight()
+
+			blocks, err := cp.blocksWithRetry(blockToRetrieve, latestBlock.Height, latestBlock.Height)
```

Comment:
why the limit is set to `latestBlock.Height`? Maybe we need a config item for poller?